src/registry/map.tsx

```
@@ -1532,6 +1531,416 @@ function MapArc<T extends MapArcDatum = MapArcDatum>({
+        return {
+          type: "Feature",
+          properties,
+          geometry: normalizePolygonGeometry(coordinates),
+        };
+      }),
+    }),
+    [data],
+  );
+
+  const latestRef = useRef({ data, onClick, onHover });
+  latestRef.current = { data, onClick, onHover };
```

ESLint

[error] Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

/home/jailuser/git/src/registry/map.tsx:1776:3
  1774 |
  1775 |   const latestRef = useRef({ data, onClick, onHover });
> 1776 |   latestRef.current = { data, onClick, onHover };
       |   ^^^^^^^^^^^^^^^^^ Cannot update ref during render
  1777 |
  1778 |   // Add source and layers on mount.
  1779 |   useEffect(() => {

(react-hooks/refs)